src/lib/UIKitConfigProvider/index.tsx

```diff
@@ -0,0 +1,121 @@
+const initialConfig = {
+  common: {
+    enableUsingDefaultUserProfile: false,
+  },
+  groupChannel: {
+    channel: {
+      enableMention: false,
+      enableOgtag: true,
+      enableReactions: true,
+      enableTypingIndicator: true,
+      enableVoiceMessage: false,
+      input: {
+        camera: {
+          enablePhoto: true,
+          enableVideo: true,
+        },
+        enableDocument: true,
+        gallery: {
+          enablePhoto: true,
+          enableVideo: true,
+        },
+      },
+      replyType: 'quote_reply',
+      threadReplySelectType: 'thread',
+    },
+    channelList: {
+      enableMessageReceiptStatus: false,
+      enableTypingIndicator: false,
+    },
+    setting: {
+      enableMessageSearch: false,
+    },
+  },
+  openChannel: {
+    channel: {
+      enableOgtag: true,
+      input: {
+        camera: {
+          enablePhoto: true,
+          enableVideo: true,
+        },
+        enableDocument: true,
+        gallery: {
+          enablePhoto: true,
+          enableVideo: true,
+        },
+      },
+    },
+  },
+};
+
+interface UIKitConfigContextInterface {
+  initDashboardConfigs: (sdk: SendbirdChat) => void;
+  configs: UIKitConfigInfo;
+}
+const UIKitConfigContext = createContext<UIKitConfigContextInterface>({
+  /* eslint-disable @typescript-eslint/no-empty-function */
+  initDashboardConfigs: () => {},
+  configs: {} as UIKitConfigInfo,
+});
+
+interface UIKitConfigProviderProps {
+  appConfigurations: DeepPartial<UIKitConfigInfo>;
+  children: React.ReactElement;
+  // If the storage value is not provided,
+  // it'll fetch the new configs set by dashboard everytime
+  storage?: any;
+  onSuccess?: () => void;
+  onError?: (error: any) => void;
+}
+
+const UIKitConfigProvider = ({ storage, children, onError, onSuccess /* appConfigurations */ }: UIKitConfigProviderProps) => {
+  // TODO: Implement
+  // const localConfigs = mergeConfigs(initialConfig, mapToUIKItConfig(configurations));
+  const [remoteConfigs, setRemoteConfigs] = useState(initialConfig);
+
+  const initDashboardConfigs = useCallback(async (sdk: SendbirdChat) => {
+    try {
+    // TODO: Implement
+    // const manager = new UIKitConfigStorageManager(storage)
+    // const storedConfigs = await manager.init(sdk.appId);
+
+      if (storage == null) {
+        const payload = await sdk.getUIKitConfiguration().json;
+        setRemoteConfigs(snakeToCamel(payload.last_updated_at));
+      }
+
+      // TODO: Implement
+      // Compare the sdk uikitConfig's lastUpdatedAt <-> stored one
+      // if(sdk.appInfo.uikitConfigInfo.lastUpdatedAt === storedConfigs.lastUpdatedAt) {
+      //   setRemoteConfigs(storedConfigs);
+      // } else {
+      //   const newConfigs = await sdk.getUIKitConfig();
+      //   const updatedConfigs = await manager.update(newConfigs);
+      //   setRemoteConfigs(updatedConfigs);
+      // }
+      onSuccess?.();
+    } catch (error) {
+      onError?.(error);
```

Comment:
It seems a bit awkward to receive the completion handler of a function executed in a different context as props and handle it. Wouldn't it be better to directly handle it when calling the function? (then/catch, try/catch, or callback)

Reply:
Yeah agreed. Here's the change :) 
https://github.com/sendbird/sendbird-uikit-react/pull/561/commits/5245991197cc8ef39fd7dfd14086c2d9474679da